pkg/archive/copy.go

```diff
@@ -151,11 +154,19 @@ func CopyInfoSourcePath(path string, followLink bool) (CopyInfo, error) {
 		return CopyInfo{}, err
 	}
 
+	var dsize int64
+	if stat.Mode().IsDir() {
+		if dsize, err = directory.Size(resolvedPath); err != nil {
+			return CopyInfo{}, err
+		}
+	}
 	return CopyInfo{
 		Path:       resolvedPath,
 		Exists:     true,
 		IsDir:      stat.IsDir(),
 		RebaseName: rebaseName,
+		Size:       stat.Size(),
+		DSize:      dsize,
```

Comment:
But, I don't want to change the behavior of Size, which maybe introduce a regression. There is a possiblilty somebody want to get only the size of a dir path rather than the all files in it.